### ISSUE TITLE
'Documentation from tiddlywiki.com' plugin cannot be searched by "standard" search

### ISSUE THREAD

**Issue by gernert** [Jermolene/TiddlyWiki5#2927]
See https://groups.google.com/d/msg/tiddlywiki/nAZ_fyOMz_M/TxkyXbnKAQAJ

When you install the 'Documentation from tiddlywiki.com' plugin ($:/plugins/tiddlywiki/tw5.com-docs) you cannot search the documentation with 'Standard search'.

You can use Advanced search, tab Shadows but this gives entries starting with `$:/` as well.

**Comment by morosanuae** [comment]
My opinion is that the documentation tiddlers should not interfere with user created tiddlers in the first place. A solution may be to have separated (dedicated) tabs in the standard/advanced search panels.

**Comment by gernert** [comment]
@morosanuae 
As given in https://groups.google.com/d/msg/tiddlywiki/nAZ_fyOMz_M/JzYrRAknAgAJ

**Comment by morosanuae** [comment]
Exactly @gernert! But I think that it deserves a dedicated tab also in the standard search. I saw something similar on Tobias Beer site (https://tobibeer.github.io/tb5/) in the standard search box. Maybe you should create an issue or pull request at least with the advanced search doc tab.

**Comment by gernert** [comment]
@morosanuae 
I think it is up to the developers how they exactly want to implement it. Th discussion is here already.

**Comment by morosanuae** [comment]
@gernert I'm not talking about the implementation details but what the casual users expect/want and I've referenced a practical example of how this can be made. I think we all (the users) have an opinion (not just developers) about the design and usability.

**Comment by pmario** [comment]
I think, that the documentation should be managed by the community. ... The problem, that we have at the moment is, that the contribution "entry bar" is a bit too high.

**Comment by morosanuae** [comment]
@pmario, I agree that the documentation should be managed by the community. I saw a pull request with a simple fix (#2859) that is lying around waiting to be commited from 14 may. I think the main "bottleneck" is that all this process is very centralized around the main "brain", Jeremy. On the other hand I did not understand what do you mean with the entry bar being to high. Is it too difficult for a regular user to learn how to make a pull request on GitHub? Even with a proper tutorial?

**Comment by pmario** [comment]
> Is it too difficult for a regular user to learn how to make a pull request on GitHub? Even with a proper tutorial?

Yes. that's exactly the problem.

**Comment by sukima** [comment]
>Is it too difficult for a regular user to learn how to make a pull request on GitHub?

This fascinates me, I've never been able to figure out why this is. A PR in GitHub is so simple to me yet I've seen so much resistance to it from others. Maybe it has to do with it is a change centric medium? I've seen experienced MS Word users get completely flustered with red and green annotations. Unfortunately, much like anything technical, the entry bar (although low) is still to high for many. It truly makes me sad to think this.

Now some projects handle the "entry bar" by having many contributors. One advantage of many OOS projects is that they have a natural level of programming competency. TiddlyWiki on the other hand often falls into the hands of non-programmers. Those who might have enough motivation to learn TiddlyWiki Text but not really the drive to learn JavaScript or how to use Git.

Another hurdle is that a PR has a lot of requirements to make it into the final output:

1.  Figure out what to change on GitHub
2.  Press edit and be forced to create an account on GitHub
3.  Verify account and become acquainted with a new system/identity.
4.  Make the change and submit a PR.
5.  Wait for feedback.
6.  Be told that they didn't sign the CLA. Which you forgot was part of the CONTRIBUTING guidelines.
7.  Read the contributing guidelines.
8.  Figure out how to sign a CLA.
9.  Make another PR with the signing of the CLA.
10. And wait for feedback again.

By this point many people would give up and move on. Especially for a small copy change. I would not blame then either. When written out it does seem rather complicated.

If I had my two cents I might suggest a lower entry where small changes did not require all of the above and then future changes are progressively shepherded to using the full GitHub interface with a signed CLA.

**Comment by morosanuae** [comment]
Can't we make a dedicated wiki (maybe one that's capable as retaining history) on a free wiki hosting service so that everybody can participate? This is a fine idea from **TonyM** [(TiddlyWikiDev)](https://groups.google.com/d/msg/tiddlywikidev/9hWr4q4Lryk/mlw7CCKXAgAJ). I've seen this article on Wikipedia (https://en.wikipedia.org/wiki/Comparison_of_wiki_hosting_services).  And instead of relying only on Jeremy as the moderator we can all vote. It will be easier and we'll have the freedom we need.
My opinion is that the documentation should be on a separate channel of development and somewhat independent (only as a plugin) so that the quick fixes should not wait until a new app release is out, a process that can take many months. A doc plugin can be updated independently of the main app. But I don't know if a plugin can be updated manually, I didn't encounter this situation yet.

**Comment by TiddlyTweeter** [comment]
@morosanuae,  the problem with diversifying via other solutions outwith what currently exists is NOT lack of will, its lack of people, I think. Look at the numbers of activists and I think you will immediately see the issue. It sees to me that this discussion is part a proxy pointer to a wider issue of getting TW better known & getting more people engaged. Of course, that is a Catch-22. And solving it is not easy.

**Comment by TiddlyTweeter** [comment]
> > Is it too difficult for a regular user to learn how to make a pull request on GitHub?

@sukima 
> This fascinates me, I've never been able to figure out why this is.

PR's LOOK different to folk like me. And actually they are. To be useful you have to be precise. To be precise you have to know what you are talking about in terms of something relating to, ultimately CODE. There is no way that the useful poetic insights I sometimes have on Google Groups can translate directly to a PR. It  involves quite a lot of work for non-programmers to think appropriately for them.

My two cents,
Josiah

**Comment by pmario** [comment]
>There is no way that the useful poetic insights I sometimes have on Google Groups can translate directly to a PR. It involves quite a lot of work for non-programmers to think appropriately for them.

hmm, do you think it's easy for devs? ... It's hard work!  Creating plugins is easy. ... 1h ...  making it usable and provide docs that somehow work for user is at least 1 day.

**Comment by TiddlyTweeter** [comment]
@pmario Completely agree with you. Its serious work.

**Comment by morosanuae** [comment]
So what's the conclusion in the end? We give up? It's fine by me if that is what everybody wants. But once again, let me tell you my plan (what I'm gone do) and you tell me if I'm talking a complete nonsense or not or if wish to give a hand or not. I'm not the ultimate expert in wiki's in general, but I offer my services to do the managing if will be the need.

**MASTERPLAN**

1.  Find a free wiki hosting service.
2. Publish my initial proposed doc structure - more details om TiddlyWikiDev forum here: https://groups.google.com/forum/#!topic/tiddlywikidev/9hWr4q4Lryk
3. We all then can participate to discussions and make contributions - I think this should be easy enough for everyone.
4. Once we a reach a certain agreed baseline, I will push the modification to GitHub.
5. The process repeats forever :)

So. it's ok or not? Do you have better solutions?

**Comment by gernert** [comment]
I opened an issue 'Documentation from tiddlywiki.com' plugin cannot be searched by "standard" search.

To my opinion the discussion drifted off topic.

**Comment by pmario** [comment]
> So what's the conclusion in the end? We give up?

... no ... go on with your ideas. 

----

For me personally TiddlyWiki docs has to be written with TiddlyWiki syntax. Since TW lacks multi editor capabilities at the moment, I invest my time to change this. So I personally don't edit eg: mediawiki content. ... but that's just me

-----------

@gernert ... you are right the docs meta discussion should be moved to a different issue.

**Comment by TiddlyTweeter** [comment]
@gernert Right. It has drifted way off your question. I'm sorry. I will pay more attention in future.

**Comment by morosanuae** [comment]
A different issue has existed not so long ago, but no one bothered to participate so... here we are.

**Comment by morosanuae** [comment]
For those who are interested we can continue the "meta" discussion on TiddlyWikiDev forum (https://groups.google.com/forum/#!topic/tiddlywikidev/9hWr4q4Lryk)

**Comment by tobibeer** [comment]
I think the simplest implementation for this is adding a dedicated search mechanism for "docs".

It should be two things, imho:

1. definitely be a tab under advanced search
1.1. which can be turned on/off in the plugin options
2. a section in the search results
2.1. which can be turned on/off in the plugin options, so that doc topics don't pop up all the time

A implementation seems rather simple, I'd think.
